crates/floresta-chain/src/pruned_utreexo/chain_state_builder.rs

```diff
@@ -19,22 +19,34 @@ use super::chainparams::ChainParams;
 use super::ChainStore;
 
 #[derive(Clone, Debug)]
+/// This enum is used to represent errors that can occur during the construction of a ChainState instance.
 pub enum BlockchainBuilderError {
+    /// Indicates that the chainstore is missing.
     MissingChainstore,
+    /// Indicates that the chain parameters are missing.
     MissingChainParams,
 }
 #[derive(Clone, Debug, Default)]
+/// Represents a builder for constructing a ChainState instance.
 pub struct ChainStateBuilder<PersistedState: ChainStore> {
+    /// The accumulator stump.
```

Comment:
I think these fields are important to have a more verbose description, since that might be the first time the consumer is reading it.